integrationtests/src/test/java/io/kroxylicious/proxy/ExpositionIT.java

```diff
@@ -74,27 +72,24 @@ public void exposesClusterOverTls(KafkaCluster cluster) throws Exception {
                 .withKeyStoreFile(brokerCertificateGenerator.getKeyStoreLocation())
                 .build();
         builder.addToVirtualClusters("demo", demo);
-        var config = builder.build().toYaml();
-
-        try (var proxy = startProxy(config)) {
-            try (var admin = CloseableAdmin.create(Map.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress,
-                    CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SSL.name,
-                    SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, clientTrustStore.toAbsolutePath().toString(),
-                    SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, brokerCertificateGenerator.getPassword()))) {
-                // do some work to ensure connection is opened
-                createTopic(admin, TOPIC, 1);
-
-                var connectionsMetric = admin.metrics().entrySet().stream().filter(metricNameEntry -> "connections".equals(metricNameEntry.getKey().name()))
-                        .findFirst();
-                assertTrue(connectionsMetric.isPresent());
-                var protocol = connectionsMetric.get().getKey().tags().get("protocol");
-                assertThat(protocol).startsWith("TLS");
-            }
+
+        try (var tester = KroxyTesters.inProcessKroxyTester(builder);
+                var admin = tester.clients("demo").admin(Map.of(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SSL.name,
```

Comment:
Maybe `Clients` could become an interface and `KroxyTester` is a `Clients` for the default vCluster, but can also be asked for a virtualCluster specific Clients. Like:

```
interface KroxyTester extends Clients {
    Clients clientsForVirtualCluster(String virtualCluster)
}
```

Reply:
maybe the Clients interface is better because KroxyTester is Closeable. it would be odd to ask KroxyTester for another tester that is also Closeable.

Reply:
I've gone with your original suggestion of using methods like `var admin = tester.admin("demo", Map.of(...))`. There are also methods without the `virtualCluster` (like `tester.admin()`) that only work in the standard case where we only have a single virtual cluster, otherwise they throw an exception about the `virtualCluster` being ambiguous.